overlord/devicestate/devicemgr.go

```diff
@@ -309,16 +309,22 @@ func (m *DeviceManager) StartUp() error {
 	return EarlyConfig(m.state, m.preloadGadget)
 }
 
-func (m *DeviceManager) maybeSetupUbuntuSave() error {
-	// only called for UC20
+func (m *DeviceManager) shouldMountUbuntuSave() bool {
+	if release.OnClassic {
+		return false
+	}
+	// TODO:UC20+: ubuntu-save needs to be mounted for recover too
+	return m.SystemMode(SysHasModeenv) == "install" ||
+		m.SystemMode(SysHasModeenv) == "run"
+}
 
+func (m *DeviceManager) maybeMountUbuntuSave() error {
```

Comment:
Shouldn't this be called just `mountUbuntuSave()`, or maybe `ensureUbuntuSaveIsMounted()`, since (unless it fails with an error) once it returns the directory will be always mounted?

Reply:
Good point!